bridges/otelzap/encoder.go

```diff
@@ -18,47 +18,69 @@ var (
 	_ zapcore.ArrayEncoder  = (*arrayEncoder)(nil)
 )
 
+type newNameSpace struct {
+	ns   string
+	kv   []log.KeyValue
+	next *newNameSpace
+}
+
 // objectEncoder implements zapcore.ObjectEncoder.
 // It encodes given fields to OTel key-values.
 type objectEncoder struct {
-	kv []log.KeyValue
+	// root is a pointer to the default namespace
+	root *newNameSpace
+	// cur is a pointer to the namespace we're currently writing to.
+	cur *newNameSpace
 }
 
-// nolint:unused
 func newObjectEncoder(len int) *objectEncoder {
 	keyval := make([]log.KeyValue, 0, len)
-
-	return &objectEncoder{
+	m := &newNameSpace{
 		kv: keyval,
 	}
+	return &objectEncoder{
+		root: m,
+		cur:  m,
+	}
+}
+
+// It iterates to the end of the linked list and appends namespace data.
+// Run this function before accessing complete result.
+func (m *objectEncoder) getEncodedResult(o *newNameSpace) {
```

Comment:
Can we rename it to something like `calculate` or `aggregate`? The current name suggests that it returns something and that it is a pure function (accessor)